### PR TITLE
IGNITE-17825 ThreadPoolMetricTest.test is flaky

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-17825

### CODE REVIEW

File: modules/metrics/src/test/java/org/apache/ignite/internal/metrics/ThreadPoolMetricTest.java [hunk 1]
```diff
@@ -54,8 +54,9 @@ public void test() throws ExecutionException, InterruptedException {
 
         assertEquals(0L, completedTaskCount.value());
 
-        exec.submit(() -> {}).get();
-        assertEquals(1L, completedTaskCount.value());
+        exec.submit(() -> {}).get(1, TimeUnit.SECONDS);
+
+        waitForCondition(() -> completedTaskCount.value() > 0, 10, TimeUnit.SECONDS.toMillis(1));
```
Comment (sashapolo) [inline modules/metrics/src/test/java/org/apache/ignite/internal/metrics/ThreadPoolMetricTest.java:59]: There should probable be an `assertTrue` here

Comment (sashapolo) [reply modules/metrics/src/test/java/org/apache/ignite/internal/metrics/ThreadPoolMetricTest.java]: Also, the code in the end should be placed in either `finally` block, or `afterEach` block

Comment (tkalkirill) [reply modules/metrics/src/test/java/org/apache/ignite/internal/metrics/ThreadPoolMetricTest.java]: Fix it. thanks

File: modules/metrics/pom.xml [hunk 2]
```diff
@@ -85,6 +85,13 @@
             <scope>test</scope>
             <type>test-jar</type>
         </dependency>
+
+        <dependency>
```
Comment (ibessonov) [inline modules/metrics/pom.xml:89]: Please reflect this change in Gradle file as well

File: modules/metrics/src/test/java/org/apache/ignite/internal/metrics/ThreadPoolMetricTest.java (hunk 1 above)
Comment (tkalkirill) [reply modules/metrics/src/test/java/org/apache/ignite/internal/metrics/ThreadPoolMetricTest.java]: Fix it all

File: modules/metrics/pom.xml (hunk 2 above)
Comment (tkalkirill) [reply modules/metrics/pom.xml:89]: Fix it

**Review by sashapolo — APPROVED** [review]

**Pull request merged**